tests/suites/test_suite_ssl.function

```diff
@@ -339,6 +341,259 @@ int mbedtls_mock_tcp_recv_nb( void *ctx, unsigned char *buf, size_t len )
     return mbedtls_test_buffer_get( socket->input, buf, len );
 }
 
+/*
+ * Structure with endpoint's certificates for SSL communication tests.
+ */
+typedef struct mbedtls_endpoint_certificate
+{
+    mbedtls_x509_crt ca_cert;
+    mbedtls_x509_crt cert;
+    mbedtls_x509_crt cert2;
+    mbedtls_pk_context pkey;
+    mbedtls_pk_context pkey2;
+} mbedtls_endpoint_certificate;
+
+/*
+ * Endpoint structure for SSL communication tests.
+ */
+typedef struct mbedtls_endpoint
+{
+        const char *name;
+        mbedtls_ssl_context ssl;
+        mbedtls_ssl_config conf;
+        mbedtls_ctr_drbg_context ctr_drbg;
+        mbedtls_entropy_context entropy;
+        mbedtls_mock_socket socket;
+        mbedtls_endpoint_certificate cert;
+} mbedtls_endpoint;
+
+#define EP( VAR )   ( &( ep -> VAR ) )
+#define CERT( VAR ) ( &( cert -> VAR ) )
+
+/*
+ * Initialize \p ep_cert structure and assigns it to endpoint
+ * represented by \p ep
+ *
+ * \retval  0 on success, otherwise error code
```

Comment:
What I meant here was listing possible errors that can be returned from this function as a separate `\retval`.

Reply:
Are you sure it is useful? Because I think analyzing what errors are possible to return in case of more complex functions it is pointless. I would have to list each function that returns any error code and, in my opinion, that kind of documentation would not be more helpful.